File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/updateendpoint/UpdateEndpointImplV0_6.java
package com.evernym.verity.sdk.protocols.updateendpoint;

import com.evernym.verity.sdk.exceptions.UndefinedContextException;
import com.evernym.verity.sdk.exceptions.VerityException;
import com.evernym.verity.sdk.protocols.AbstractProtocol;
import com.evernym.verity.sdk.protocols.updateendpoint.common.Authentication;
import com.evernym.verity.sdk.protocols.updateendpoint.v0_6.UpdateEndpointV0_6;
import com.evernym.verity.sdk.utils.Context;
import org.json.JSONArray;
import org.json.JSONObject;

import java.io.IOException;

/*
 * NON_VISIBLE
 *
 * This is an implementation of UpdateEndpointImplV0_6 but is not viable to user of Verity SDK. Created using the
 * static UpdateEndpoint class
 */
class UpdateEndpointImplV0_6 extends AbstractProtocol implements UpdateEndpointV0_6 {

    Authentication authentication = null;

    UpdateEndpointImplV0_6() {
        super();
    }

    UpdateEndpointImplV0_6(Authentication authentication) {
        super();
        this.authentication = authentication;
    }

    final int COM_METHOD_TYPE = 2;

    @Override
    public void update(Context context) throws IOException, VerityException {
        send(context, updateMsg(context));
    }

    @Override
    public JSONObject updateMsg(Context context) throws UndefinedContextException {
        JSONObject message = new JSONObject();

        message.put("@type", messageType(UPDATE_ENDPOINT));
        message.put("@id", getNewId());
        JSONObject comMethod = new JSONObject();
        comMethod.put("id", "webhook");
        comMethod.put("type", COM_METHOD_TYPE);
        comMethod.put("value", context.endpointUrl());
        JSONObject packaging = new JSONObject();
        packaging.put("pkgType", "1.0");
        JSONArray recipientKeys = new JSONArray();
        recipientKeys.put(context.sdkVerKey());
        packaging.put("recipientKeys", recipientKeys);
        comMethod.put("packaging", packaging);
        if (authentication != null)
            comMethod.put("authentication", authentication.toJson());
        message.put("comMethod", comMethod);
        return message;
    }

    @Override
    public byte[] updateMsgPacked(Context context) throws VerityException {
        return packMsg(context, updateMsg(context));
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/UpdateEndpoint/UpdateEndpointImplV0_6.cs
using System;
using System.Collections.Generic;
using System.Json;
using System.Text;

namespace VeritySDK
{
    /// <summary>
    /// This is an implementation of UpdateEndpointImplV0_6 but is not viable to user of Verity SDK. Created using the static UpdateEndpoint class
    /// </summary>
    public class UpdateEndpointImplV0_6 : UpdateEndpointV0_6
    {
        public UpdateEndpointImplV0_6() : base() { }

        int COM_METHOD_TYPE = 2;

        public override void update(Context context)
        {
            send(context, updateMsg(context));
        }

        public override JsonObject updateMsg(Context context)
        {
            JsonObject message = new JsonObject();

            message.Add("@type", messageType(UPDATE_ENDPOINT));
            message.Add("@id", getNewId());

            JsonObject comMethod = new JsonObject();
            comMethod.Add("id", "webhook");
            comMethod.Add("type", COM_METHOD_TYPE);
            comMethod.Add("value", context.EndpointUrl());

            JsonObject packaging = new JsonObject();
            packaging.Add("pkgType", "1.0");
            
            JsonArray recipientKeys = new JsonArray();
            recipientKeys.Add(context.SdkVerKey());
            packaging.Add("recipientKeys", recipientKeys);

            comMethod.Add("packaging", packaging);

            message.Add("comMethod", comMethod);
            return message;
        }

        public override byte[] updateMsgPacked(Context context)
        {
            return packMsg(context, updateMsg(context));
        }
    }
}
